### ISSUE TITLE
[CMake] Change the layout of the builded files inside the cmake-build directory

### ISSUE THREAD

**Issue by ice0** [synfig/synfig#1131]
Check the directory structure which is created by 2-build-debug.sh or 2-build-release.sh
Modify the CMake scripts to make the same structure.

**Comment by Keyikedalube** [comment]
Last time I tried 2-build-cmake.sh on Fedora the built files were stored in a folder named something *portable(?)*
After switching to Debian 2-build-cmake.sh is not usable [#928](https://github.com/synfig/synfig/issues/928#issuecomment-575589239) So don't correctly remember how builded files are stored for cmake.

But here's how I understand this issue. You want to replace _debug/ and _production/ directory depending on arg passed to build-cmake and not create a separate build directory?

**Comment by ice0** [comment]
Here current CMake status. (https://github.com/synfig/synfig/issues/928)

And commands you need to execute:
```
git clone https://github.com/synfig/synfig
cd synfig
mkdir build-cmake && cd build-cmake
cmake ..  -DCMAKE_INSTALL_PREFIX=~/synfig-cmake
make -j4
make install
export PATH=~/synfig-cmake/bin/:$PATH
export LD_LIBRARY_PATH=~/synfig-cmake/lib/:$LD_LIBRARY_PATH
export SYNFIG_ROOT=~/synfig-cmake/
make build_images -j4
make install
# fix incorrect install path for icons
mkdir ~/synfig-cmake/share/pixmaps/synfigstudio
mv ~/synfig-cmake/share/pixmaps/*.png ~/synfig-cmake/share/pixmaps/synfigstudio
```

As you can see, you need to do `make install` before `make build_images`.
If you try to run `synfig` (console binary needed for build_images target), it doesn't start, because it cannot find `libsynfig.so` in it's build directory. And to do that, we run `make install` command and it creates `bin` and `lib` directories, and binary can be executed at this time.
The goal of this issue is to do the same thing, but not at `install` step, but right at build step. This will allow to user, just do `cmake ..` to do the whole build.

**Comment by ice0** [comment]
Directory structure, in CMake terminology:

```
${PROJECT_BINARY_DIR}/${CMAKE_BUILD_TYPE}/bin
${PROJECT_BINARY_DIR}/${CMAKE_BUILD_TYPE}/etc
${PROJECT_BINARY_DIR}/${CMAKE_BUILD_TYPE}/lib
...
and so on
```

**Comment by Keyikedalube** [comment]
Now I understand that cmake-build will build into its own custom build directory and not replace _debug/ or _production/ directory
*That's what I thought this issue was addressing: use the same debug and production build directory with CMake by using the same folder layout. I understood it wrong*

The issue is correcting the build script to make `synfig` ready for building images without invoking `make install` prematurely

Thanks for the insight @ice0 :)
I'm learning CMake just recently. Might tinker around synfig CMake build script just to get more exposure and gain experience ;P

**Comment by AYESDIE** [comment]
As pointed on the forums, I made changes to some of the `CMakeLists.txt` files to match the structure of the directory created by `2-build-debug.sh`.

**Comment by ice0** [comment]
Fixed by #1224
